p5/io/http_data.py

```diff
@@ -1,29 +1,38 @@
-from urllib import request, parse
+"""
+Part of the io library. Includes functions that require **internet** connection.
+"""
+
+from urllib import request
 import json
 import requests
 
-# synchronous
-
+# Synchronous
 
-def http_get(url):
+class http_get:
     """
     Performs a synchronous GET request on the given url.
 
     :param url: url of the API
     :type url: string
 
     """
-    try:
-        resp = request.urlopen(url)
-    except Exception:
-        print("Invalid url. Connection Failed.")
-        exit(1)
-    data = resp.read()
-    data = json.loads(data.decode("UTF-8"))
-    return data
 
-# synchronous
+    def __init__(self, url):
+        try:
+            resp = request.urlopen(url)
+        except:
+            print("Invalid url. Connection Failed.")
+            exit(1)
+
+        data = resp.read().decode("UTF-8")
+
+        self.text = data
+        try:
+            self.json = json.loads(data)
+        except: 
```

Comment:
When do we expect `json.loads()` to fail?

Reply:
Well if the resp is 
```dart
test
```
then fail or 
```json
{wrong syntax
```